src/shipper.py

```diff
@@ -26,34 +32,61 @@ class UnknownURL(Exception):
 
 
 class LogzioShipper(object):
-    MAX_BULK_SIZE_IN_BYTES = 1 * 1024 * 1024
+    MAX_BULK_SIZE_IN_BYTES = 3 * 1024 * 1024
 
     def __init__(self, logzio_url):
-        self._size = 0
-        self._logs = []
+        self._decompress_size = 0
+        self._logs = StringIO.StringIO()
         self._logzio_url = logzio_url
+        self._logs_counter = 0
+        try:
+            self._compress = os.environ['COMPRESS'].lower() in ["yes", "true", "t", "1"]
+        except KeyError:
+            self._compress = False
+        self._writer = self._create_new_writer()
+
+    def _create_new_writer(self):
+        self._logs.truncate(0)
```

Comment:
if you'll want to refactor that later for multi-writer or something i think that this will produce a hard to locate bug.